src/main/scala/io/iohk/ethereum/domain/package.scala

```diff
@@ -2,20 +2,29 @@ package io.iohk.ethereum
 
 import akka.util.ByteString
 import io.iohk.ethereum.mpt.{ByteArrayEncoder, ByteArraySerializable, HashByteArraySerializable, MerklePatriciaTrie, NodesKeyValueStorage}
-import io.iohk.ethereum.rlp.UInt256RLPImplicits._
+import io.iohk.ethereum.rlp.RLPImplicits._
+import io.iohk.ethereum.utils.ByteUtils
+import org.spongycastle.util.BigIntegers
 
 package object domain {
 
-  val byteArrayUInt256Serializer = new ByteArrayEncoder[UInt256] {
-    override def toBytes(input: UInt256): Array[Byte] = input.bytes.toArray[Byte]
+  val byteArrayBigIntSerializer = new ByteArrayEncoder[BigInt] {
+    override def toBytes(input: BigInt): Array[Byte] =
+      ByteUtils.padLeft(ByteString(BigIntegers.asUnsignedByteArray(input.bigInteger)), 32).toArray[Byte]
   }
 
-  val rlpUInt256Serializer = new ByteArraySerializable[UInt256] {
-    override def fromBytes(bytes: Array[Byte]): UInt256 = ByteString(bytes).toUInt256
-    override def toBytes(input: UInt256): Array[Byte] = input.toBytes
+  val rlpBigIntSerializer = new ByteArraySerializable[BigInt] {
+    override def fromBytes(bytes: Array[Byte]): BigInt = {
+      BigIntegers.fromUnsignedByteArray(rlp.decode[ByteString](bytes).toArray[Byte])
```

Comment:
We already have an RLP codec for BigInts (RLPImplicits.scala)

Reply:
...and it works for nonnegative values only. I see no point, I think it can be changed

Reply:
I've been debugging ETS for the past hour and I was pretty sure using exisitng BigInt encoder was the cause for failures. Let me double check that

Reply:
OK, it seems the failures were because of missing padding in key serializer. I've changed it back to use existing BigInt encoder here